Add = as Operator in Puppet lexer

alexharv074 opened the pull request:
The = sign had been missed apparently as a valid Puppet operator.
Fixes #979.

dblessing commented on spec/lexers/puppet_spec.rb:
```diff
@@ -3,6 +3,55 @@
 describe Rouge::Lexers::Puppet do
   let(:subject) { Rouge::Lexers::Puppet.new }
 
+  describe 'lexing' do
```
Thanks for writing some tests. This isn't necessary, though. Simply adding an example in `samples/puppet` is sufficient.

dblessing commented on lib/rouge/lexers/puppet.rb:
```diff
@@ -64,6 +64,13 @@ def self.metaparameters
+        # interpreting /= as the divide-and-assign operator!
+        #
+        rule /\=/, Operator
+
```
Would it make sense to add `=` to the `rule %r([<>!+*/-]), Operator` regex below? This would also likely have the added benefit of not needing the leading slash and the comment wouldn't be needed

alexharv074 commented:
@dblessing , I refactored to avoid the confusing syntax as you suggested. Meanwhile, I left the Rspec tests in. I think they assist any future maintainers to get their heads around why a 2013-era syntax highlighter actually works on the latest version of Puppet?

dblessing commented:
@gfx @jneen Do you have an opinion on the tests? We have some similar tests in the code-base but I recall @jneen telling me once that they aren't needed like this.

dblessing commented:
> I think they assist any future maintainers to get their heads around why a 2013-era syntax highlighter actually works on the latest version of Puppet?

FWIW Rouge is constantly updated and lexers are added, updated, etc. as needed. When Rouge was created has no bearing one whether it does or doesn't support the latest version of some language.

jneen commented:
I'd generally prefer adding the example to `spec/samples/puppet`, since I've found the usualy-style tests to be quite brittle when we have to accomodate other language changes.

alexharv074 commented:
@dblessing @jneen  for what it's worth, I found it much easier to analyse the behaviour of the lexers using the Rspec framework than I did using the debug output from `rougify` so an understanding of these tests should help maintainers develop their code. But with that said, I removed the tests and added sample code in `spec/visual/samples/puppet` as requested. Those tests an be restored by checking out e1ad61a48e59d810a810c747ae8ba1e0967ac855.

alexharv074 commented:
@dblessing not sure if you're waiting on me here, but I can't find any way from my end to move this Issue out of "Changes requested" state. Let me know if there's anything else I am supposed to to?

alexharv074 commented:
Any chance this could be merged now @dblessing ?

alexharv074 commented:
@dblessing Looks like this was merged over 6 months ago but still my GitHub/Jekyll blog doesn't have the fix. Any idea why that might be?

dblessing commented:
Make sure to update dependencies. Looks like this was released in Rouge
3.3.0.
Alex Harvey <notifications@github.com> schrieb am Mo. 25. März 2019 um
01:58:

> @dblessing <https://github.com/dblessing> Looks like this was merged over
> 6 months ago but still my GitHub/Jekyll blog doesn't have the fix. Any idea
> why that might be?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jneen/rouge/pull/980#issuecomment-476077067>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ACSKlNqLdl6XcOqGo5SXEIrg4biPj3szks5vaHOOgaJpZM4WYqbe>
> .
>

alexharv074 commented:
Yeah, I just specify latest jekyll on my side and the rest happens magically on the GitHub side. Ok, thanks for merging and releasing anyway! I'll investigate.
